src/mlpack/methods/reinforcement_learning/training_config.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file training_config.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is the implementation of TrainingConfig class,
+ * which contains hyper-parameters for training RL agent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+#define MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+
+namespace mlpack {
+namespace rl {
+
+class TrainingConfig
```

Comment:
`CV` shouldn't be a problem, as I don't think there exists some `cross validation` in RL. Maybe in the future I need to refactor the API to make it consistent with other parts of mlpack, but it won't block the `CV` project anyway for now. As for hyper-parameter tuner, I'm not very clear how it works and it seems in that discussion there hasn't be a clear standard. 

> In the hyperparameter tuner it only makes sense for the user to pass all of the hyperparameters to the Optimize() function

RL does need a hyper parameter tuner, but if we need to pass all the hyper parameters to `Train()`, it may be a disaster for user.